File: app/src/main/java/simple/clever/notes/data/CardSource.java
package simple.clever.notes.data;

public interface CardSource {
    CardSource init(CardSourceResponse cardSourceResponse);
    CardData getCardData(int position);
    int size();
    void addCardData(CardData cardData);
    void deleteCardData(int position);
    void updateCardData(CardData cardData, int position);


}

File: app/src/main/java/simple/clever/notes/data/CardSource.kt
package simple.clever.notes.data

interface CardSource {
    fun init(cardSourceResponse: CardSourceResponse): CardSource
    fun getCardData(position: Int): CardData
    fun size(): Int
    fun addCardData(cardData: CardData)
    fun deleteCardData(position: Int)
    fun updateCardData(cardData: CardData,position: Int)


}
